### PR TITLE
Apache APISIX HelmChart Initial commit

### PR DESCRIPTION
**Pull request by shenal**
- This PR is the Initial commit of the Apache APISIX HelmChart

### CODE REVIEW

**Comment by membphis** [comment]
@gxthrj you can take a look at this PR too.

File: chart/apisix/templates/tests/test-connection.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  labels:
+    {{- include "apisix.labels" . | nindent 4 }}
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['wget']
+      args: ['{{ include "apisix.fullname" . }}:{{ .Values.gateway.http.port }}']
```
Comment (gxthrj) [inline chart/apisix/templates/tests/test-connection.yaml:30]: No definition found `.Values.gateway.http.port `
```suggestion
      args: ['{{ include "apisix.fullname" . }}:{{ .Values.gateway.http.containerPort }}']
```

File: chart/apisix/values.yaml [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        paths: []
+    tls: []
+  #  - secretName: apisix-tls
+  #    hosts:
+  #      - chart-example.local
+
+# etcd configuration
+# use the FQDN address or the IP of the etcd
+etcd:
+  host: http://etcd-apisix.local:2379 # http://172.20.128.89:2379
```
Comment (gxthrj) [inline chart/apisix/values.yaml:81]: > http://etcd-apisix.local:2379
 Do we need some explanation about this  address? Or we need an etcd deployment definition?

File: chart/apisix/values.yaml [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+  # Overrides the image tag whose default is the chart appVersion.
+  tag: 1.3-alpine
+
+imagePullSecrets: []
+nameOverride: ""
+fullnameOverride: ""
+
+serviceAccount:
+  # Specifies whether a service account should be created
+  create: true
```
Comment (gxthrj) [inline chart/apisix/values.yaml:30]: I think `serviceaccount` is not necessary, should this place be set to `false`?

File: chart/apisix/values.yaml [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+  #   drop:
+  #   - ALL
+  # readOnlyRootFilesystem: true
+  # runAsNonRoot: true
+  # runAsUser: 1000
+
+gateway:
+  type: LoadBalancer
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-type: nlb
```
Comment (gxthrj) [inline chart/apisix/values.yaml:53]: It seems too special here, users may not apply `aws`.
It is recommended to use the `NodePort` type by default, which is more general.

File: chart/apisix/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        checksum/config: {{ include (print $.Template.BasePath "/configmap.yaml") . | sha256sum }}
+    {{- end }}
+      labels:
+        {{- include "apisix.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "apisix.serviceAccountName" . }}
```
Comment (gxthrj) [inline chart/apisix/templates/deployment.yaml:44]: I think `serviceaccount` is not necessary, Maybe we can remove this line :)

File: chart/apisix/templates/ingress.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{{- if .Values.gateway.ingress.enabled -}}
+{{- $fullName := include "apisix.fullname" . -}}
+{{- $svcPort := .Values.service.port -}}
```
Comment (gxthrj) [inline chart/apisix/templates/ingress.yaml:19]: It seems that ingress can't work yet. Will it be improved in the future?

File: chart/apisix/values.yaml (hunk 2 above)
Comment (shenal) [reply chart/apisix/values.yaml:81]: @gxthrj I have added a comment above to specify the ETCD FQDN

Comment (shenal) [reply chart/apisix/values.yaml:81]: If we can add an ETCD chart as a dependency as a separate improvement

File: chart/apisix/templates/deployment.yaml (hunk 5 above)
Comment (shenal) [reply chart/apisix/templates/deployment.yaml]: Removed the SA

File: chart/apisix/values.yaml (hunk 4 above)
Comment (shenal) [reply chart/apisix/values.yaml]: Done

File: chart/apisix/values.yaml (hunk 2 above)
Comment (gxthrj) [reply chart/apisix/values.yaml:73]: > If we can add an ETCD chart as a dependency as a separate improvement

Yes, I think it's better,We can improve it in the future !

File: chart/apisix/templates/ingress.yaml (hunk 6 above)
Comment (shenal) [reply chart/apisix/templates/ingress.yaml]: Fixed

**Comment by membphis** [comment]
@shenal @gxthrj @sshniro merged first

**Pull request merged**